main.py

```diff
@@ -58,13 +58,14 @@ def create_chip_file(chip_obj, file_name):
 # Notes:        Second argument MUST BE LEFT TO DEFAULT!!!
 #               It is used internally, as a saved version of
 #               original chip argument
+#               Partial credit to: @bill090
 def create_new_chip(chip, return_chip = None):
     builtin_components = ["AND", "NOT", "SIGNAL IN", "SIGNAL OUT"]
     other_components = []
 
-    # if its the first iteration of recursion then return chip = chip
-    if return_chip == None:

```

Comment:
Wait why did you remove this? this was making it so that if this function was called by the user and not by recursion then the return chip would equal the input chip so we could modify it more afterwards

Reply:
The recursion happens regardless of the return_chip input.

Reply:
But you can bring it back and do more with it,